OptimizelySDK/Odp/OdpManager.cs

```diff
@@ -0,0 +1,274 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+using OptimizelySDK.ErrorHandler;
+using OptimizelySDK.Logger;
+using OptimizelySDK.Odp.Entity;
+using System;
+using System.Collections.Generic;
+
+namespace OptimizelySDK.Odp
+{
+    /// <summary>
+    /// Concrete implementation to orchestrate segment manager, event manager, and ODP config
+    /// </summary>
+    public class OdpManager : IOdpManager
+    {
+        /// <summary>
+        /// Denotes if ODP Manager is meant to be handling ODP communication
+        /// </summary>
+        private bool _enabled;
+
+        /// <summary>
+        /// Configuration used to communicate with ODP
+        /// </summary>
+        private volatile OdpConfig _odpConfig;
+
+        /// <summary>
+        /// Manager used to handle audience segment membership
+        /// </summary>
+        public IOdpSegmentManager SegmentManager { get; private set; }
+
+        /// <summary>
+        /// Manager used to send events to ODP 
+        /// </summary>
+        public IOdpEventManager EventManager { get; private set; }
+
+        /// <summary>
+        /// Logger used to record messages that occur within the ODP client
+        /// </summary>
+        private ILogger _logger;
+
+        /// <summary>
+        /// Update the settings being used for ODP configuration and reset/restart dependent processes
+        /// </summary>
+        /// <param name="apiKey">Public API key from ODP</param>
+        /// <param name="apiHost">Host portion of the URL of ODP</param>
+        /// <param name="segmentsToCheck">Audience segments to consider</param>
+        /// <returns>True if settings were update otherwise False</returns>
+        public bool UpdateSettings(string apiKey, string apiHost, List<string> segmentsToCheck)
+        {
+            var newConfig = new OdpConfig(apiKey, apiHost, segmentsToCheck);
+            if (_odpConfig.Equals(newConfig))
+            {
+                return false;
+            }
+
+            _odpConfig = newConfig;
+
+            EventManager.Flush();
+            EventManager.UpdateSettings(_odpConfig);
+
+            SegmentManager.ResetCache();
+            SegmentManager.UpdateSettings(_odpConfig);
+
+            return true;
+        }
+
+        /// <summary>
+        /// Attempts to fetch and return a list of a user's qualified segments.
+        /// </summary>
+        /// <param name="userId">FS User ID</param>
+        /// <param name="options">Options used during segment cache handling</param>
+        /// <returns>Qualified segments for the user from the cache or the ODP server</returns>
+        public List<string> FetchQualifiedSegments(string userId, List<OdpSegmentOption> options)
+        {
+            if (!_enabled || SegmentManager == null || !_odpConfig.IsReady())
```

Comment:
a minor comment, 
` SegmentManager == null || !_enabled || !_odpConfig.IsReady()`

Reply:
I can re-arrange (I think that's what you're asking)  

I thought that perhaps `!_enabled` would evaluate to true more often than the others, but happy to switch.